docs/contributing/style_guide.rst

```diff
@@ -1,2 +1,64 @@
 Style Guide
 ===========
+
+Python Version
+--------------
+All Pulp 3+ code will use Python 3.5+. It is not necessary to maintain backwards compatibility
+with Python 2.Y.
+
+PEP-8
+-----
+All code should be compliant with PEP-8_ where reasonable.
+
+It is recommended that contributors check for compliance by running flake8_. We include
+``flake8.cfg`` files in our git repositories for convenience.
+
+.. _PEP-8: https://www.python.org/dev/peps/pep-0008
+.. _flake8: http://flake8.pycqa.org/en/latest/
+
+Modifications:
+**************
+line length: We limit to 100 characters, rather than 79.
+
+In-code Documentation
+---------------------
+Most classes and functions should have a docstring that follows the conventions described in
+`Google's Python Style Guide <https://google.github.io/styleguide/pyguide.htmlshowone=Comments#Comments>`_.
+
+Exceptions and Clarifications
+*****************************
+#. Modules should not include license information.
+#. The type of each Args value should be included after the variable name in parentheses. The type of each Returns value should be the first item on the line.
+#. Following the type of Args and Returns values, there will be a colon and a single space followed by the description. Additional spaces should not be used to align types and descriptions.
+#. Fields and Relations sections will be used when documenting fields on Django models. The Fields section will be used for non-related fields on Model classes, the Relations section will be used for related fields on Model classes.
```

Comment:
The second sentence is really two, so you can end with "on Model classes. The Relations section..."